### PR TITLE
Cleanup modules and dependencies

### PR DESCRIPTION
**Pull request by ddworak**
- removed obscure testing modules
- replaced joda-time
- dependency updates

### CODE REVIEW

File: scex-util/src/main/scala/com/avsystem/scex/util/EnrichedDate.scala [hunk 1]
```diff
@@ -49,27 +50,27 @@ final class EnrichedDate(private val wrapped: Date) extends AnyVal {
...
-  def dayOfWeek: Int = dateTime.getDayOfWeek
+  def dayOfWeek: Int = instant.get(ChronoField.DAY_OF_WEEK)
 
   @Documentation("Returns a numeric value for the day of the year.")
-  def dayOfYear: Int = dateTime.getDayOfYear
+  def dayOfYear: Int = instant.get(ChronoField.DAY_OF_YEAR)
 
   @Documentation("Returns a numeric value for the month of the year, where 1 - January, 12 - December.")
-  def monthOfYear: Int = dateTime.getMonthOfYear
+  def monthOfYear: Int = instant.get(ChronoField.MONTH_OF_YEAR)
```
Comment (sebaciv) [inline scex-util/src/main/scala/com/avsystem/scex/util/EnrichedDate.scala:74]: Unfortunately:
```
Unsupported field: MonthOfYear
java.time.temporal.UnsupportedTemporalTypeException: Unsupported field: MonthOfYear
	at java.base/java.time.Instant.get(Instant.java:567)
	at com.avsystem.scex.util.EnrichedDate$.monthOfYear$extension(EnrichedDate.scala:74)
	at com.avsystem.scex.util.EnrichedDateTest.$anonfun$new$7(EnrichedDateTest.scala:44)
```

File: scex-util/src/main/scala/com/avsystem/scex/util/function/FormatUtilImpl.java [hunk 2]
```diff
@@ -22,25 +20,12 @@ public class FormatUtilImpl implements FormatUtil {
     private static final Pattern numericPattern = Pattern.compile(NUMERIC_PATTERN);
 
     public static String getPeriodMessage(Date from) {
```
Comment (sebaciv) [inline scex-util/src/main/scala/com/avsystem/scex/util/function/FormatUtilImpl.java:22]: Seems unused, remove?

File: scex-util/src/main/scala/com/avsystem/scex/util/EnrichedDate.scala (hunk 1 above)
Comment (sebaciv) [reply scex-util/src/main/scala/com/avsystem/scex/util/EnrichedDate.scala:74]: Consider adding tests for these methods

**Review by sebaciv — CHANGES_REQUESTED** [review]

File: build.sbt [hunk 3]
```diff
@@ -23,16 +23,13 @@ val CompileAndTest = "compile->compile;test->test"
 
 val parserCombinatorsVersion = "2.4.0"
 val avsCommonsVersion = "2.16.0"
```
Comment (sebaciv) [inline build.sbt:25]: Add commons 2.17.0 as well

File: scex-util/src/main/scala/com/avsystem/scex/util/function/FormatUtilImpl.java (hunk 2 above)
Comment (ddworak) [reply scex-util/src/main/scala/com/avsystem/scex/util/function/FormatUtilImpl.java:22]: There was a usage in UMP but it's easy enough to remove, sure.

File: scex-util/src/main/scala/com/avsystem/scex/util/EnrichedDate.scala (hunk 1 above)
Comment (ddworak) [reply scex-util/src/main/scala/com/avsystem/scex/util/EnrichedDate.scala:74]: Done. Note that the tests depend on the timezone, so I fixed it to Europe/Warsaw.

File: scex-util/src/main/scala/com/avsystem/scex/util/EnrichedDate.scala [hunk 4]
```diff
@@ -1,20 +1,25 @@
...
+import java.util.{Calendar, Date}
+
+final class EnrichedDate(wrapped: Date, zone: ZoneId = Clock.systemDefaultZone().getZone) {
+  private def zonedDateTime: ZonedDateTime = ZonedDateTime.ofInstant(Instant.ofEpochMilli(wrapped.getTime), zone)
 
   @Documentation("Formats the date using the default date format: <tt>yyyy.MM.dd HH:mm:ss</tt>.")
-  def format: String = CommonDateFormat.get.format(wrapped)
+  def format: String =
+    if (zone == Clock.systemDefaultZone().getZone)
+      CommonDateFormat.get.format(wrapped)
```
Comment (ddworak) [inline scex-util/src/main/scala/com/avsystem/scex/util/EnrichedDate.scala:17]: Not sure how important that optimization was

**Review by sebaciv — CHANGES_REQUESTED** [review]

File: scex-util/src/main/scala/com/avsystem/scex/util/EnrichedDate.scala [hunk 5]
```diff
@@ -1,20 +1,25 @@
...
-import org.joda.time.DateTime
 
-final class EnrichedDate(private val wrapped: Date) extends AnyVal {
-  private def dateTime = new DateTime(wrapped.getTime)
+import java.time.format.DateTimeFormatter
+import java.time.temporal.ChronoField
+import java.time.{Clock, Instant, ZoneId, ZonedDateTime}
+import java.util.{Calendar, Date}
+
+final class EnrichedDate(wrapped: Date, zone: ZoneId = Clock.systemDefaultZone().getZone) {
```
[the remark this answers is not in the document]
Comment (ddworak) [reply scex-util/src/main/scala/com/avsystem/scex/util/EnrichedDate.scala:11]: On the other hand: a single cached `DateTimeFormatter` is immutable, thread-safe and around 5200000 ops/s when cached globally, so I may be tempted to remove the optimization. WDYT?

**Comment by ddworak** [comment]
Not converting `com.avsystem.scex.util.StringMiscOps` and `com.avsystem.scex.util.CommonExpressionUtils#stringToDateOnSetter` to new formatter intentionally - they are using `java.util.Date` as target type so this would get ugly.

**Review by sebaciv — APPROVED** [review]

**Pull request merged**